lang/c++/impl/DataFile.cc

```diff
@@ -306,20 +306,20 @@ void DataFileReaderBase::init(const ValidSchema& readerSchema)
 
 static void drain(InputStream& in)
 {
-    const uint8_t *p = 0;
+    const uint8_t *p = nullptr;
     size_t n = 0;
     while (in.next(&p, &n));
 }
 
 char hex(unsigned int x)
 {
-    return x + (x < 10 ? '0' :  ('a' - 10));
+    return static_cast<char>(x + (x < 10 ? '0' :  ('a' - 10)));
 }
 
 std::ostream& operator << (std::ostream& os, const DataFileSync& s)
 {
-    for (size_t i = 0; i < s.size(); ++i) {
-        os << hex(s[i] / 16)  << hex(s[i] % 16) << ' ';
+    for (unsigned char i : s) {
```

Comment:
it looks like elsewhere uint8_t might be used of rhte same type?

Reply:
You're right. Fixed